### PR TITLE
Improve troubleshooting guide for issues switching from Docker Engine

### PR DESCRIPTION
**Pull request by NitishKumar06**
Fix: #34756

**Note**
The approach that I followed was to add a small paragraph at the beginning of the page (https://kubernetes.io/docs/tasks/debug/debug-cluster/) **Cluster not starting** which will guide the users to navigate to the following page (https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/) to understand more about the error: `Error: failed to parse kubelet flag: unknown flag: --network-plugin`

Also, Before upgrading to Kubernetes 1.24, it is essential to go through [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes) . This could also be seen from @sftim [comment](https://github.com/kubernetes/website/issues/34756#issue-1291383994) on the issue that this PR is fixing.

Please feel free to suggest changes if necessary.

### CODE REVIEW

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:28]: We tend to use relative link instead of full URI.

```suggestion
*Note: If you get the following error after upgrading to Kubernetes 1.24, go through the [Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/) page to get a better understanding on how to resolve this error.*
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:17]:
```suggestion
 Before upgrading to Kubernetes 1.24 , please go through [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes).
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:22]:
```suggestion
To start your cluster, run the following command:
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:28]:
```suggestion
*Note: If you get the following error after upgrading to Kubernetes 1.24, go through the [Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/) guide to get a better understanding of how to resolve this error.*
```

**Review by divya-mohan0209 — DISMISSED** [review]
Few grammatical nits + incorporated suggestions from @Sea-n

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (kbhawkey) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:17]: nit: You can remove the word, 'please'.
Before you upgrade to Kubernetes 1.24, review the [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes).

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:28]:
```suggestion
If you get the following error after upgrading to Kubernetes 1.24, and you were previously using Docker Engine as your container runtime, go through the [Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/) guide.
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:30]: This is _not_ a shell script. Omit `shell` here.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:26]: This is *not* a command that starts your cluster

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:26]: Thanks @sftim , I'll push the new changes now.

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md]: @sftim The new changes have been pushed. I've deleted the following command which is wrong and believe it can go without mentioning on "How to start a cluster" since it's error resolving page and not "guide page"

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:26]: I'd move this to be a level-3 heading / section within “Cluster failure modes”, and make the heading be similar to one of:

- “Retaining old settings after cluster upgrade”
- “Incomplete upgrade from Kubernetes v1.23 or earlier”
- “Node configuration wrong after switching from Docker Engine container runtime”

What do you think @NitishKumar06 ?

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:26]: This sounds good @sftim . Correct me if I'm wrong: 

So we can just transfer the above newly added information under "CLUSTER FAILURE MODES" using level-3 heading like [Specific scenarios](https://kubernetes.io/docs/tasks/debug/debug-cluster/#contributing-causes) with the above heading title which suits best. Right?

**Comment by sftim** [comment]
/retitle Improve troubleshooting guide for issues switching from Docker Engine

**Comment by sftim** [comment]
To me, https://github.com/kubernetes/website/pull/34764#discussion_r917473631 is only barely a nit. It really would be better to remove “please”; that word isn't right for a troubleshooting guide.

However, this change is much more right than wrong.
/lgtm

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:319]: I don't think the title is in line with the content that comes along with it. Per [this guide linked in the issue](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes), the flags have been removed and I'm assuming we're trying to signpost the users to check-out the link mentioned in the content to show it has been removed. 

If that's the case, I think we'd need to rephrase the title better or introduce it with the note/caution admonition because we're not retaining anything. WDYT?

```suggestion
### Retaining old settings after cluster upgrade
```

Comment (sftim) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: @divya-mohan0209 what was the text change you wanted to suggest?

Comment (divya-mohan0209) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: So, IMO, instead of having this as a separate subsection it'd be better to signpost users with a caution/note admonition. Or we could have it retitled to "Things to note after dockershim migration" or something on those lines. I'd be curious to hear others' thoughts on this, as well.

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: @sftim The title mentioned by @divya-mohan0209 sounds appealing to me as well. Should I commit new changes?

Comment (sftim) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: You're welcome to make changes here @NitishKumar06

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: Perfect! I'll raise the new changes. Thanks @divya-mohan0209

**Review by sftim — COMMENTED** [review]
@NitishKumar06 what are your intentions for moving this work forward?

**Comment by NitishKumar06** [comment]
@sftim I've pushed the changes which @divya-mohan020 suggested and I found those changes really required looking at some of the minor typos. I think this PR is ready to be merged. WDYT?

**Comment by sftim** [comment]
I'm happy for any English reviewer to check this PR and see if it is ready for merge.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:319]:
```suggestion
### Things to note after dockershim migration
```

Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:319]: Alright , I'll push the changes as required.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:17]: Not sure why there are weird spaces.
```suggestion
Before upgrading to Kubernetes 1.24, please go through [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes).
```

And please address [previous suggestion](https://github.com/kubernetes/website/pull/34764#pullrequestreview-1033704600) to this line 4 weeks ago.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:20]: There should be a newline before and after `<!-- body -->` tag.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:28]: This issue is still not addressed.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:33]: We don't need two empty line for most cases.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:330]: Add a newline before heading.

**Comment by Sea-n** [comment]
Hey @NitishKumar06,

Could you review all the previous suggestions again? There are some issues didn't get addressed for 4 weeks.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:330]: Got it!

**Comment by NitishKumar06** [comment]
@Sea-n Yes! Please give me some time and analyse the changes well:) Thank you!

**Comment by NitishKumar06** [comment]
@Sea-n Done!

**Comment by NitishKumar06** [comment]
Any update @sftim @divya-mohan0209 @Sea-n ?

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:17]:
```suggestion
Before upgrading to Kubernetes 1.24, please go through [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes).
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:31]: Copying from @sftim:

> This is not a shell script. Omit `shell` here.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (Sea-n) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:333]: Use `{{< note >}}` shortcode instead.

**Review by sftim — COMMENTED** [review]
I wonder if we should make this change to the v1.24 docs, and for more recent releases, link back to this section specifically in the v1.24 docs?

The docs for v1.25, v1.26 etc shouldn't really talk about upgrading to v1.24&hellip;

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:333]: Do you mean I need to put the text inside `< >` ?

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:17]: DONE!

**Comment by NitishKumar06** [comment]
> I wonder if we should make this change to the v1.24 docs, and for more recent releases, link back to this section specifically in the v1.24 docs?
> 
> The docs for v1.25, v1.26 etc shouldn't really talk about upgrading to v1.24…

I agree with your suggestions @sftim . I'd recommend if we go as per your suggestion, we should first of all make sure what are the changes to be made finalised before raising a new PR to commit changes in some other branch. Hence, we can keep this PR for suggestions if we go with your idea. Once the changes are made finalised, I can close this PR and raise  a new PR in the required branch.

**Comment by sftim** [comment]
> I agree with your suggestions @sftim . I'd recommend if we go as per your suggestion, we should first of all make sure what are the changes to be made finalised before raising a new PR to commit changes in some other branch. Hence, we can keep this PR for suggestions if we go with your idea. Once the changes are made finalised, I can close this PR and raise a new PR in the required branch.

/hold

That approach is fine. (It's also OK to target the 1.24 release branch directly).

**Comment by Sea-n** [comment]
Hey @NitishKumar06,

It's been a while since the last update, should we close this PR?

If you want to change the target branch to `dev-1.24`, it's not necessary to open a new PR, just see [this tutorial](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request).

**Comment by NitishKumar06** [comment]
Hey @Sea-n @sftim , Do you have further suggestions to go for? I believe we can consider this PR as approved if no one has much suggestions regarding the same.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (reylejano) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:27]: This is incorrect
You do not start your cluster with `kubelet --help | grep network-plugin`

/hold
Holding until this is corrected

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (reylejano) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:333]: Read https://kubernetes.io/docs/contribute/style/hugo-shortcodes/ on how to use hugo shortcodes

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (reylejano) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:18]: The current `main` branch is for 1.25
I'm also ok with this line not being in the top of this page as there is no other content/context around troubleshooting upgrades. 
Maybe a warning like this (sans version) can go in the [Upgrade a Cluster](https://kubernetes.io/docs/tasks/administer-cluster/cluster-upgrade/). A similar sentence is already in the [Upgrading kubeadm clusters](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/) page

```suggestion
Before upgrading to Kubernetes 1.24, please go through [Urgent Upgrade Notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes).

```

**Comment by reylejano** [comment]
> If you want to change the target branch to dev-1.24, it's not necessary to open a new PR, just see [this tutorial](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request).

The target branch for the 1.24 docs is the `release-1.24` branch

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md:18]: @reylejano Thank you for the suggestions! I'd be committing the changes soon.

**Comment by NitishKumar06** [comment]
@reylejano Please checkout the file changes in the latest commit.

**Comment by NitishKumar06** [comment]
@sftim @reylejano @Sea-n Kindly take a look at the changes now.

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:320]: IMO, this could be phrased better. WDYT?
```suggestion
*Note: Starting 1.24, since management of the CNI is no longer in scope for kubelet, the `--network-plugin` command line parameter has been removed from the binary and will throw the following error, if used:
```

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (divya-mohan0209) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:324]: WDYT?
```suggestion
To understand how to resolve this error, go through the [Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/) guide.
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Left some suggestions. WDYT?

File: content/en/docs/tasks/debug/debug-cluster/_index.md
Comment (NitishKumar06) [reply content/en/docs/tasks/debug/debug-cluster/_index.md]: As per @reylejano suggested [here](https://github.com/kubernetes/website/pull/34764#discussion_r984924005), I've omitted this line.

**Comment by kbhawkey** [comment]
Hello @NitishKumar06 .
I think it makes sense to close this pull request. You can always create another PR with updated changes. Thanks.
/close

**Comment by NitishKumar06** [comment]
How does it make sense to open a new PR? This PR already had the suggested changes pushed. We were looking forward to getting a review from the maintainers before proceeding further rather than closing this one.

**Comment by kbhawkey** [comment]
Hello @NitishKumar06 . You can always reopen a pull request. Do you want to mark this PR as a draft or are the changes ready to merge?

**Review by tengqm — COMMENTED** [review]
I'm not sure why we have removed the heading 2 at line 29. 
I'm not sure why we add a heading 3 at line 317.
Line 319-325 is a note, that is fine. But line 319 and 325 are marked heading 2?
Why line 320 starts with a '*'?
Can we wrap the lines inside the note?

**Comment by NitishKumar06** [comment]
This workspace kind of looks messed up - Hence created a new PR: https://github.com/kubernetes/website/pull/39625

**Pull request closed without merging**
